src/Parser/Mage/Fire/Modules/Features/Pyroclasm.js

```diff
@@ -0,0 +1,209 @@
+    if (spellId !== SPELLS.PYROBLAST.id || this.beginCastFound === false) {
+      return;
+    }
+    this.beginCastFound = false;
+    debug && console.log("Pyroblast Casted @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    this.castTimestamp = event.timestamp;
+    const castTime = this.castTimestamp - this.beginCastTimestamp;
+    //Checks the begincast and cast timestamps to determine if it is instant cast or not. This doesnt matter for ABT and ToS because hot streak pyroblasts dont have a begincast, but in Nighthold they do. So this needs to remain for backwards compatibility
+    if (castTime >= CAST_BUFFER && this.combatants.selected.hasBuff(SPELLS.PYROCLASM_BUFF.id)) {
+      this.isBuffed = true;
+      this.buffUsed = true;
+      debug && console.log("Buff Used @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id) {
+      return;
+    }
+      if (this.isBuffed === true) {
+        this.damage += getDamageBonus(event, BONUS_DAMAGE);
+        this.isBuffed = false;
+        debug && console.log("Pyroblast Damage @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      } else {
+        debug && console.log("Non Buffed Pyroblast Damage @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      }
+  }
+
+  on_finished() {
+    if (this.combatants.selected.hasBuff(SPELLS.PYROCLASM_BUFF.id)) {
+      const adjustedFightEnding = this.owner.currentTimestamp - 7500;
+      if (this.buffAppliedTimestamp < adjustedFightEnding) {
+        this.wastedProcs += 1;
+        debug && console.log("Fight Ended with Unused Proc @ " + formatMilliseconds(this.owner.currentTimestamp - this.owner.fight.start_time));
+      } else {
+        this.totalProcs -= 1;
+      }
+    }
+    debug && console.log("Total Procs: " + this.totalProcs);
+    debug && console.log("Used Procs: " + this.usedProcs);
+    debug && console.log("Wasted Procs: " + this.wastedProcs);
+  }
+
+  get avgBonusDamage() {
+    return this.damage / this.usedProcs;
+  }
+
+  get procsPerMinute() {
+    console.log(this.owner.fightDuration / 60000);
+    return this.totalProcs / (this.owner.fightDuration / 60000);
+  }
+
+  get procUtilization() {
+    return 1 - (this.wastedProcs / this.totalProcs);
+  }
+
+  get procUtilizationThresholds() {
+    return {
+      actual: this.procUtilization,
+      isLessThan: {
+        minor: 0.95,
+        average: 0.90,
+        major: 0.80,
+      },
+      style: 'percentage',
+    };
+  }
+
+  suggestions(when) {
+    when(this.procUtilizationThresholds)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<React.Fragment>You wasted {formatNumber(this.wastedProcs)} of your <SpellLink id={SPELLS.PYROCLASM_TALENT.id} /> procs. Try to use your procs as soon as possible to avoid this.</React.Fragment>)
```

Comment:
“to avoid this” it would be slightly more informative to explain *why* the procs should be used, what does using more achieve?